### PR TITLE
Cleanup texmanager.

### PR DESCRIPTION
**Pull request by anntzer**
- remove check for "old" cache (the location changed in efa9ea6 (2005)).
- postscriptd and pscnt are unused since c0d6110 (2006), deprecate them.
- calling \usepackage more than once with the same package does nothing
  (as a repeated import in Python) so no need to remove calls to
  \usepackage{type1cm}.
- replace DEBUG by _log.debug where appropriate; other cases (forcing
  file regeneration) can be handled by manually deleting the old file.

Various other minor cleanups.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by tacaswell** [comment]
get_ps_bbox was last used in 310e707ed29eee5e4a5f7f4dc75b3a508a50aeac (2009)

**Comment by tacaswell** [comment]
It looks like `DEBUG` used to also short-circuit the cache and force re-generation in all cases.  I don't think that is a big deal, but just making sure it is noted.

File: lib/matplotlib/texmanager.py [hunk 1]
```diff
@@ -158,48 +133,43 @@ class TexManager(object):
...
-                        print('family: %s, font: %s, info: %s' %
-                              (font_family, font,
-                               self.font_info[font.lower()]))
+                    _log.debug('family: %s, font: %s, info: %s',
+                               font_family, font, self.font_info[font.lower()])
                     break
                 else:
-                    if DEBUG:
-                        print('$s font is not compatible with usetex')
+                    _log.debug('$s font is not compatible with usetex')
```
Comment (QuLogic) [inline lib/matplotlib/texmanager.py]: What is a `$s font`? Was this supposed to be a `%s` substitution?

File: lib/matplotlib/texmanager.py [hunk 2]
```diff
@@ -208,21 +178,16 @@ def __init__(self):
...
-        cmd = '\n'.join(cmd)
-        self._font_preamble = '\n'.join(['\\usepackage{type1cm}', cmd,
-                                         '\\usepackage{textcomp}'])
+        self._font_preamble = '\n'.join(
+            [r'\usepackage{type1cm}'] + cmd + [r'\usepackage{textcomp}'])
 
     def get_basefile(self, tex, fontsize, dpi=None):
         """
-        returns a filename based on a hash of the string, fontsize, and dpi
+        Returns a filename based on a hash of the string, fontsize, and dpi.
```
Comment (QuLogic) [inline lib/matplotlib/texmanager.py]: Since you're changing them all, the first line of the docstring should be imperative (Return, not Returns).

**Comment by anntzer** [comment]
all comments are handled

**Pull request merged**